### PR TITLE
68 refactor user to use username instead of user

### PR DESCRIPTION
**Pull request by larsuzh**

### CODE REVIEW

File: findme-location-riddles/src/LocationRiddlesService.py [hunk 1]
```diff
@@ -93,37 +95,37 @@ def get_location_riddles_for_user(
...
-        following_users = self.__get_following_users_list(event, user_id)
+        following_users = self.__get_following_users_list(event, username)
 
         response = []
         for following_user in following_users:
             try:
                 response.extend(
                     self.get_location_riddles_for_user(
-                        user_id=following_user["user_id"], requester_id=user_id
+                        username=following_user["username"], requester_id=username
```
Comment (flyingbeat) [inline findme-location-riddles/src/LocationRiddlesService.py:107]: I would rename `requester_id` to `requester`

File: findme-location-riddles/tests/test_location_riddle_service.py [hunk 2]
```diff
@@ -60,12 +61,12 @@ def test_guess_location_riddle(self):
             guess_result["location_riddle"]["location_riddle_id"],
             "mock_location_riddle_id",
         )
-        self.assertEqual(guess_result["location_riddle"]["user_id"], "mock_user_id")
+        self.assertEqual(guess_result["location_riddle"]["username"], "mock_username")
         self.assertEqual(guess_result["location_riddle"]["comments"], [])
         self.assertEqual(
-            guess_result["location_riddle"]["guesses"],
-            [{"guess": [0.0, 0.0], "user_id": "mock_user2_id"}],
+            guess_result["location_riddle"]["guesses"][0]["username"], "mock_user2_id"
```
Comment (flyingbeat) [inline findme-location-riddles/tests/test_location_riddle_service.py:67]: would rename to `mock_username2` consider also in L58

File: findme-location-riddles/tests/test_location_riddle_service.py [hunk 3]
```diff
@@ -91,7 +92,7 @@ def test_rate_location_riddle(self):
             "mock_location_riddle_id", "mock_user2_id", 3
```
Comment (flyingbeat) [inline findme-location-riddles/tests/test_location_riddle_service.py:92]: there is still some inconsistency in the tests, lets remove all references to `user_id`

File: findme-location-riddles/src/entities/Rating.py [hunk 4]
```diff
@@ -2,5 +2,5 @@
 
 
 class Rating(BaseModel):
-    user_id: str
+    username: str
     rating: int = Field(..., ge=1, le=5)
```
Comment (flyingbeat) [inline findme-location-riddles/src/entities/Rating.py:6]: is the initial rating 1? or 0?

File: serverless.yml [hunk 5]
```diff
@@ -250,21 +250,12 @@ resources:
             AttributeType: S
           - AttributeName: username
             AttributeType: S
-          - AttributeName: user_id
-            AttributeType: S
         BillingMode: PAY_PER_REQUEST
         KeySchema:
           - AttributeName: partition_key
```
Comment (flyingbeat) [inline serverless.yml:255]: is this still correct? do we still need this attribute partition_key USER? I was not sure... @larsuzh

**Review by flyingbeat — CHANGES_REQUESTED** [review]

File: serverless.yml (hunk 5 above)
Comment (larsuzh) [reply serverless.yml:255]: Yes, this should be working. (If I understood it correctly) We write it to the DB in this way in order to distinguish between users and follow requests, which are also in this table

**Review by flyingbeat — APPROVED** [review]

**Pull request merged**